packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -294,34 +294,7 @@ module.exports = {
       throw new ApplicationError('Impossible to find the default role');
     }
 
-    const { email, username, provider } = params;
-
-    const identifierFilter = {
-      $or: [
-        { email: email.toLowerCase() },
-        { username: email.toLowerCase() },
-        { username },
-        { email: username },
-      ],
-    };
-
-    const conflictingUserCount = await strapi.query('plugin::users-permissions.user').count({
-      where: { ...identifierFilter, provider },
-    });
-
-    if (conflictingUserCount > 0) {
-      throw new ApplicationError('Email or Username are already taken');
-    }
-
-    if (settings.unique_email) {
```

Comment:
Same as the comment above, the changes don't take into consideration the U&P configuration options such as unique_email.
https://docs.strapi.io/user-docs/latest/settings/configuring-users-permissions-plugin-settings.html#configuring-advanced-settings